endoreg_db/services/video_import.py

```
@@ -158,181 +269,561 @@ def _create_or_retrieve_video_instance(self):
+    def _get_processor_roi_info(self):
+        """Get processor ROI information for masking."""
+        processor_roi = None
+        endoscope_roi = None
+
+        video = self._require_current_video()
+
+        try:
+            video_meta = getattr(video, "video_meta", None)
+            processor = getattr(video_meta, "processor", None) if video_meta else None
+            if processor:
+                assert isinstance(processor, EndoscopyProcessor), "Processor is not of type EndoscopyProcessor"
```

Pylint (4.0.0)

[error] Using variable 'EndoscopyProcessor' before assignment

(E0601)